### ISSUE TITLE
Block transactions not imported in Ropsten network

### ISSUE THREAD

**Issue by mlalitthapa** [ethereum/go-ethereum#18036]
#### System information

Geth version: `v1.7.3-stable-4bb3c89d`
OS & Version: Linux
Commit hash : (if `develop`)

#### Expected behaviour
`eth.blockNumer` returns `4375556` and all the transaction from that block should be imported.

#### Actual behaviour
When I try to get the transaction information with `eth.getTransaction` from one of the transaction in block `4375556` then it returns `null`.

#### Steps to reproduce the behaviour
First I ran the geth with command `geth --testnet --syncMode "fast" --cache 1024` by creating a `service file` in Ubuntu. But when the transaction informations were not showing then I restarted the service. After some google search I restarted the service again with `geth --testnet --syncMode "full" --cache 1024` but still no transactions were imported.

#### Backtrace

````
geth.service - Geth
   Loaded: loaded (/etc/systemd/system/geth.service; disabled; vendor preset: enabled)
   Active: active (running) since Tue 2018-11-06 05:05:23 UTC; 39min ago
 Main PID: 14511 (geth)
    Tasks: 13
   Memory: 3.0G
      CPU: 1min 58.218s
   CGroup: /system.slice/geth.service
           └─14511 /usr/bin/geth --testnet --syncmode full --cache 1024

Nov 06 05:43:17 ip-10-10-10-138 geth[14511]: INFO [11-06|05:43:17] Imported new chain segment               blocks=1  txs=0  mgas=0.000 elapsed=3.297ms   mgasps=0.000   number=4375571 hash=50948a…c96717
Nov 06 05:43:26 ip-10-10-10-138 geth[14511]: INFO [11-06|05:43:26] Imported new chain segment               blocks=1  txs=28 mgas=4.483 elapsed=19.937ms  mgasps=224.830 number=4375572 hash=4bbc07…e0f5e5
Nov 06 05:43:31 ip-10-10-10-138 geth[14511]: INFO [11-06|05:43:31] Imported new chain segment               blocks=1  txs=0  mgas=0.000 elapsed=3.378ms   mgasps=0.000   number=4375573 hash=a1466d…63f375
Nov 06 05:43:33 ip-10-10-10-138 geth[14511]: INFO [11-06|05:43:33] Imported new chain segment               blocks=1  txs=0  mgas=0.000 elapsed=3.685ms   mgasps=0.000   number=4375574 hash=91a449…7c63d8
Nov 06 05:43:34 ip-10-10-10-138 geth[14511]: INFO [11-06|05:43:34] Imported new chain segment               blocks=1  txs=2  mgas=0.081 elapsed=5.572ms   mgasps=14.538  number=4375574 hash=a43432…52c814
Nov 06 05:43:53 ip-10-10-10-138 geth[14511]: INFO [11-06|05:43:53] Imported new chain segment               blocks=1  txs=0  mgas=0.000 elapsed=3.333ms   mgasps=0.000   number=4375575 hash=2fe949…e0c264
Nov 06 05:44:03 ip-10-10-10-138 geth[14511]: INFO [11-06|05:44:03] Imported new chain segment               blocks=1  txs=1  mgas=0.021 elapsed=8.222ms   mgasps=2.554   number=4375576 hash=a2aad9…c36f5d
Nov 06 05:44:22 ip-10-10-10-138 geth[14511]: WARN [11-06|05:44:22] Ancestor below allowance                 peer=3c9c542b970ed183 number=4285576 hash=000000…000000 allowance=4285576
Nov 06 05:44:22 ip-10-10-10-138 geth[14511]: WARN [11-06|05:44:22] Synchronisation failed, dropping peer    peer=3c9c542b970ed183 err="retrieved ancestor is invalid"
Nov 06 05:44:51 ip-10-10-10-138 geth[14511]: INFO [11-06|05:44:51] Imported new chain segment               blocks=1  txs=0  mgas=0.000 elapsed=3.418ms   mgasps=0.000   number=4375577 hash=1804f1…6a84f4
````
This is the log for `service geth status`. You can see that some of the blocks were imported with `txs=0`. Is it the expected behaviour? When will the transactions be imported?

**Comment by karalabe** [comment]
Please provide a log where you do an `eth.getBlock("0x...")` listing the block details, and then do one for the transaction itself. It's a bit hard to follow your repro because you're missing the actual calls that fail.

**Comment by holiman** [comment]
Actually, `v1.7.3-stable-4bb3c89d` means you're running a non-constantinople enabled node, so whatever you're getting will most certainly _not_ be the same as what https://ropsten.etherscan.io/block/4375556 shows, because you're on different forks.

**Comment by mlalitthapa** [comment]
Yes, I think @holiman is right. I updated my geth to `v1.8` and started syncing with bootnodes available at https://github.com/ethereum/ropsten and now everything seems to work fine.

**Comment by karalabe** [comment]
Thanks for confirming it's fixed!
